### ISSUE TITLE
Unable to enable OpenID config

### ISSUE THREAD

**Issue by PKizzle** [minio/minio#17500]
## Expected Behavior
Using the web console or `mc` it should be possible to enable an OpenID provider configured using the official helm chart.

## Current Behavior
The following error is thrown:
```
Unable to remove openid IDP config '_': 'config_url' is not optional for 'identity_openid' sub-system, please check 'identity_openid' documentation.
```

## Steps to Reproduce (for bugs)
1. Install MinIO using the helm chart and configure all necessary options in the `oidc` section
2. Launch MinIO while the OpenID provider is offline to disable this configuration
3. Try to enable the OpenID configuration after the provider is back online using the web console or `mc idp openid enable ${ALIAS}`

## Context
I want to automatically enable the OpenID configuration as soon as the provider is available. As I am self-hosting it I can not guarantee 100% uptime. MinIO only checks for the availability on startup, so I want to use `mc` in a script to enable the OpenID configuration once it becomes available.

## Regression
Unknown. All of the last 3 releases on quay.io had this problem.

## Your Environment

* Version used (`minio --version`): RELEASE.2023-06-19T19-52-50Z
* Server setup and configuration: MinIO helm chart 5.0.10 on a 1.27.2 k3s cluster (3x RaspberryPi 4B)
* Operating System and version (`uname -a`): Linux 6.1.21-v8+ aarch64 GNU/Linux

**Comment by donatello** [comment]
The error message indicates that the server never loaded the configuration in the first place (so there is nothing to "enable"). However we did find a bug in the server init.

This should fix your issue, https://github.com/minio/minio/pull/17544 - but for this use case, I suggest setting the openid configuration via environment variables:

1. If the openid auth server is not available at minio startup time, the configuration fails validation and is not added.
2. If the openid server is available, all is good, except if it goes down later, login attempts fail, but the configuration remains loaded in the server.

In case of 1, a simple restart of minio will load the openid configuration.

If you prefer to do it via mc, you have to check the error message content because the server differentiates between `mc idp openid add|update`. Only if the server successfully loads your openid config, you will be able to later enable or disable it.

**Comment by PKizzle** [comment]
Is there also an option to enable the openid configuration in case of 1 without performing a full restart (no downtime)?

**Comment by harshavardhana** [comment]
> Is there also an option to enable the openid configuration in case of 1 without performing a full restart (no downtime)?

There is no downtime in HTTP APIs, they all reconnect once they reconnect upon a connection disconnect.

This is not a Postgres, think of MinIO like nginx restart.

**Comment by PKizzle** [comment]
Sorry in advance for my stupid follow up question.
Do you mean that any client will intelligently retry to connect to the API should it not be available (downtime)?

**Comment by donatello** [comment]
> Is there also an option to enable the openid configuration in case of 1 without performing a full restart (no downtime)?

No. The first time minio loads openid config, it attempts to check if the openid service is up and do some initialization. If it is down, the configuration is considered invalid and not applied. Only a restart retries the whole process.

If minio loads the configuration at startup successfully, i.e. able to connect to auth service etc, then openid based login should work whenever the auth service is up.

**Comment by harshavardhana** [comment]
Fixed by #17544
